### ISSUE TITLE
no-var incorrectly autofixes declarations in statement position

### ISSUE THREAD

**Issue by not-an-aardvark** [eslint/eslint#7961]
**Tell us about your environment**

* **ESLint Version:** master
* **Node Version:** 7.4.0
* **npm Version:** 4.0.5

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

none

**What did you do? Please include the actual source code causing the issue.**

```js
/* eslint no-var: error */

if (foo) var bar = 1;
```

**What did you expect to happen?**

I expected an error to be reported, but no fix to be performed.

**What actually happened? Please include the actual, raw output from ESLint.**

The code was fixed to

```js
if (foo) let bar = 1;
```

which is a syntax error.
